library_generation/utilities.sh

```diff
@@ -76,7 +76,9 @@ remove_grpc_version() {
 download_gapic_generator_pom_parent() {
   local gapic_generator_version=$1
   if [ ! -f "gapic-generator-java-pom-parent-${gapic_generator_version}.pom" ]; then
-    if [[ "${gapic_generator_version}" == *"-SNAPSHOT" ]]; then
+    # release please branches will point to a non-snapshot that hasn't been
+    # published. We want to get the generator pom locally in this case
+    if [[ "${gapic_generator_version}" == *"-SNAPSHOT" ]] || [[ "$(is_release_branch)" == "true" ]]; then
```

Comment:
Can we always try to download from local maven repo first regardless of the name, and fall back to maven central if not exist? We are going to have multiple release branches in the future and this would become hard to maintain. 
Similarly, we would have the same problem in the [main generation script](https://github.com/googleapis/sdk-platform-java/blob/main/library_generation/utilities.sh#L123-L137), can we try to fix it there as well @JoeWang1127 ?

Reply:
thanks for the quick review! 
I modified the generator download (pom & jar) to default to a local fetch before trying the external source